test/embag_test.cc

```diff
@@ -227,6 +227,31 @@ TEST_F(ViewTest, MessagesForTopic) {
   }
 }
 
+TEST_F(ViewTest, MessagesBetweenTimestamps) {
+  std::chrono::nanoseconds start_time_ns {view_.getStartTime().to_nsec()};
+  std::chrono::nanoseconds end_time_ns   {view_.getEndTime().to_nsec()};
+  start_time_ns += std::chrono::seconds{1};
+
+  std::vector<std::string> all_topics{
+      "/base_pose_ground_truth",
+      "/base_scan",
+      "/luminar_pointcloud",
+  };
+
+  int no_time_boundaries_count = 0;
+  for (const auto &message : view_.getMessages(all_topics)){
+      no_time_boundaries_count++;
+  }
+
+  int with_time_boundaries_count = 0;
+  for (const auto &message : view_.getMessages(all_topics, start_time_ns, end_time_ns)) {
+      with_time_boundaries_count++;
```

Comment:
We should also likely assert that the timestamp of the message is greater than or equal to the start time, and less than the end_time

Reply:
@liamembark we can't actually - this isn't true. Maybe you can suggest how we can document what is happening:

This compares **chunks** and not actual messages. So it is entirely possible (and happens in this case) that if i say give me messages between t=5 and t=11, we get messages from t=4 to t=12 because those are the chunks that encompass those messages.

Reply:
nvm see below